core/tests/test_trezor.crypto.slip39.py

```diff
@@ -97,63 +102,57 @@ def test_group_sharing_threshold_1(self):
         member_thresholds = (3, 2, 2, 1)
         identifier = slip39.generate_random_identifier()
         mnemonics = slip39.generate_mnemonics_from_data(
-            self.EMS, identifier, group_threshold, list(zip(member_thresholds, group_sizes))
+            self.EMS, identifier, group_threshold, list(zip(member_thresholds, group_sizes)), 1
         )
 
         # Test all valid combinations of mnemonics.
         for group, threshold in zip(mnemonics, member_thresholds):
             for group_subset in combinations(group, threshold):
                 mnemonic_subset = list(group_subset)
                 random.shuffle(mnemonic_subset)
-                identifier, exponent, ems = slip39.combine_mnemonics(mnemonic_subset)
+                identifier, exponent, ems, group_count = slip39.combine_mnemonics(mnemonic_subset)
+                self.assertEqual(group_count, len(group_sizes))
                 self.assertEqual(ems, self.EMS)
 
 
     def test_all_groups_exist(self):
         for group_threshold in (1, 2, 5):
             identifier = slip39.generate_random_identifier()
             mnemonics = slip39.generate_mnemonics_from_data(
-                self.EMS, identifier, group_threshold, [(3, 5), (1, 1), (2, 3), (2, 5), (3, 5)]
+                self.EMS, identifier, group_threshold, [(3, 5), (1, 1), (2, 3), (2, 5), (3, 5)], 1
             )
             self.assertEqual(len(mnemonics), 5)
             self.assertEqual(len(sum(mnemonics, [])), 19)
 
 
     def test_invalid_sharing(self):
         identifier = slip39.generate_random_identifier()
-        # Short master secret.
-        with self.assertRaises(ValueError):
-            slip39.generate_mnemonics_from_data(self.EMS[:14], identifier, 1, [(2, 3)])
```

Comment:
why are these two tests gone?

Reply:
I believe we have removed these two exceptions: https://github.com/trezor/trezor-firmware/pull/539/files#diff-4238181ae22ca803d6f2de869476ab99L509-L519

Reply:
hm AFAICT the reason is that we have weakened the checks in `generate_mnemonics_from_data` because we don't need them